romancal/resample/resample.py

```diff
@@ -1,318 +1,220 @@
-        self.output_filename = output
-        self.pscale_ratio = pscale_ratio
-        self.single = single
-        self.pixfrac = pixfrac
-        self.kernel = kernel
-        self.fillval = fillval
-        self.weight_type = wht_type
-        self.good_bits = good_bits
-        self.in_memory = kwargs.get("in_memory", True)
-        if "target" in input_models.asn:
-            self.location_name = input_models.asn["target"]
-        else:
-            self.location_name = "None"
-
-        log.info(f"Driz parameter kernel: {self.kernel}")
-        log.info(f"Driz parameter pixfrac: {self.pixfrac}")
-        log.info(f"Driz parameter fillval: {self.fillval}")
-        log.info(f"Driz parameter weight_type: {self.weight_type}")
-
-        output_wcs = kwargs.get("output_wcs", None)
-        output_shape = kwargs.get("output_shape", None)
-        crpix = kwargs.get("crpix", None)
-        crval = kwargs.get("crval", None)
-        rotation = kwargs.get("rotation", None)
-
-        if pscale is not None:
-            log.info(f"Output pixel scale: {pscale} arcsec.")
-            pscale /= 3600.0
-        else:
-            log.info(f"Output pixel scale ratio: {pscale_ratio}")
-
-        # build the output WCS object
-        if output_wcs:
-            # use the provided WCS object
-            self.output_wcs = output_wcs
-            if output_shape is not None:
-                self.output_wcs.array_shape = output_shape[::-1]
-        else:
-            with self.input_models:
-                models = list(self.input_models)
-                # determine output WCS based on all inputs, including a reference WCS
-                self.output_wcs = resample_utils.make_output_wcs(
+
+        self._compute_exptime = compute_exptime
+        self._blend_meta = blend_meta
+
+        if output_wcs is None:
+            if wcs_kwargs is None:
+                wcs_kwargs = {}
+            with input_models:
+                models = list(input_models)
+                wcs = make_output_wcs(
                     models,
-                    pscale_ratio=self.pscale_ratio,
-                    pscale=pscale,
-                    rotation=rotation,
-                    shape=None if output_shape is None else output_shape[::-1],
-                    crpix=crpix,
-                    crval=crval,
+                    **wcs_kwargs,
                 )
                 for i, m in enumerate(models):
-                    self.input_models.shelve(m, i, modify=False)
-
-        log.debug(f"Output mosaic size: {self.output_wcs.array_shape}")
-
-        # n_images sets the number of context image planes.
-        # This should be 1 to start (not the default of 2).
-        self.blank_output = maker_utils.mk_datamodel(
-            datamodels.MosaicModel, n_images=1, shape=tuple(self.output_wcs.array_shape)
+                    input_models.shelve(m, i, modify=False)
+
+            # FIXME use s_regions here, but this causes data differences
```

Comment:
See https://github.com/spacetelescope/romancal/issues/1633

Reply:
Yeah that's the plan.
- update the s_regions
- use the s_regions here
- remove `make_output_wcs` (and other code no longer needed)